### ISSUE TITLE
GIS Basemap

### ISSUE THREAD

**Issue by kaichi0521** [domlysz/BlenderGIS#655]
macOS Monterey 12.3.1  

# **Blender and OS versions**

# **Describe the bug**

# **How to Reproduce**

# **Error message**

**Comment by wittywit** [comment]
### Blender and OS versions
macOS Monterey 12.3.1
MacBook Pro M1 14 inch
Blender 3.0.0 

### Describe the bug
I have the same issue I have installed the addon. The cache folder also has a GOOGLE_SAT_WM.gpkg file. But no map loads from basemap option. I even waited for 10 minutes still nothing. Downgrading the blender to 2.93 LTS also same result it did not work at all. 

<img width="1510" alt="image" src="https://user-images.githubusercontent.com/15814315/163223480-5d3b47c7-07e8-4414-8373-73c9df07a6a6.png">


### How to Reproduce
I believe just use any mac. maybe a mac specific bug. 
### Error message

no error messages popup.

**Comment by wittywit** [comment]
### Update

I downgraded the Blender even further to 2.83 LTS and the addon now works it showcases the map.

**Comment by Skytale23** [comment]
macOS Monterey 12.3.1
M1 mini (M1, 2020)

Similar issue - using Blender 2.93, 3.0 and 3.1 all for Apple Silicon: basemap not loading, Blender frozen - force quit to exit.
So downloaded Blender 3.1.2 for Intel, installed Addon, selected area, E key downloaded map to plane, then GIS-Webgeo Data-Get Elevation (SRTM)-Server: Marine Geo.org - worked
Other servers ie OpenTopography SRTM returned error.

**Comment by wittywit** [comment]
For OpenTopography SRTM error you should do the following :

The solution was also discussed here: https://github.com/domlysz/BlenderGIS/issues/619

- Create an account at: https://portal.opentopography.org/ 
- Generate a new API: Go to https://portal.opentopography.org/lidarAuthorizationInfo and select or create a new API key
- Replace your API Key in the following URL: https://portal.opentopography.org/API/globaldem?demtype=SRTMGL1&west={W}&east={E}&south={S}&north={N}&outputFormat=GTiff&API_Key=**_[YOUR API KEY ]_**

- Replace this URL in blender addon settings: 
<img width="1250" alt="Screenshot 2022-04-24 at 12 37 49 AM" src="https://user-images.githubusercontent.com/15814315/164942970-3be8e73b-6642-4ee5-a0be-57bc38886fe3.png">

**Comment by Timotheus747** [comment]
@Skytale23 You got it to work in 3.1? I couldn't get it to show the map, so downgraded to 3.0 and it worked fine.

**Comment by aplr** [comment]
TL;DR: imageio lib is outdated, temp. workaround:
```
brew install freeimage
launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib
```
cheers.

Long story, lots of learnings:

So, I dug in a bit and found, that the error message was a symptom, but not the root cause. The original error was shadowed, and is thrown in thrown [here](https://github.com/domlysz/BlenderGIS/blob/7acdaae88ddcb43345daca9e4b310fd8bc626ce4/core/lib/imageio/plugins/_freeimage.py#L408). This error simply means: “The freeimage library could not be loaded.” So, the imageio library can't find the freeimage library it depends upon. Also, it can't download the library, as the version of freeimage is outdated and does not know about the existence of an arm macos. So even if you have installed freeimage on your system, imageio won’t find it.

So, what can we do? Luckily, I found [this block of code](https://github.com/domlysz/BlenderGIS/blob/7acdaae88ddcb43345daca9e4b310fd8bc626ce4/core/lib/imageio/plugins/_freeimage.py#L465), which gave hope.
```
# Add the path specified with IMAGEIO_FREEIMAGE_LIB:
lib = os.getenv('IMAGEIO_FREEIMAGE_LIB', None)
if lib is not None:
    exact_lib_names.insert(0, lib)
```

I installed the freeimage library using `brew install freeimage`, there’s an arm build also.

Then, setting the env var in the terminal and starting blender did the job (yay 🥳). It’s not the most convenient solution, as you always have to start blender from the terminal or some custom script, even if you set the env var in `~/.zshrc` or something.

Fortunately, you can set env vars for gui applications with `launchctl`, so it’s as easy as running the following:

```
launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib
```

Just make sure you’re entering the correct version, it was `3.18.0` at the time of writing.

cheers again.

**Comment by sbusso** [comment]
that works, thanks @aplr

**Comment by danielhamilton** [comment]
Thanks, @aplr. Can confirm that it also works on MacOS 12.5.1, Blender 3.3.0, BlenderGIS 2.2.8

**Comment by liurenjie** [comment]
`launchctl setenv IMAGEIO_FREEIMAGE_LIB ...`
The environment-variables setting will out of work after the MacOS reboot.

**Comment by tabrs** [comment]
Thenks a lot! Helped Mac M1 Pro, MacOS 12.7, Blender 3.3

> ### TL;DR
> 
> imageio lib is outdated, temp. workaround:
> 
> ```
> brew install freeimage
> launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib
> ```
> 
> _running on macOS 12.3, blender 3.2.1, blendergis 2.2.8_

**Comment by leonitousconforti** [comment]
Thank you @aplr! Still works perfectly on macos 13.0, Blender 3.5.1, BlenderGIS 2.2.8

**Comment by multitudes** [comment]
It did work for me for a while. 
Now I get this error again! 
I have Blender 3.5.1 and Ventura 13.4

I did install - updated again freeimage and launched the env. 
```bash
brew install freeimage
launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib
```
Then I launch Blender from the terminal.

and I get the dreaded : "ValueError: Could not find a format to write the specified file in mode 'I'"

Could be a Python binding problem?

Device with name Apple M1 Pro supports metal minimum requirements
METAL API - DETECTED GPU: Apple M1 Pro
DEBUG:BlenderGIS-master.core.checkdeps:GDAL Python binding unavailable
DEBUG:BlenderGIS-master.core.checkdeps:PyProj unavailable
DEBUG:BlenderGIS-master.core.checkdeps:Pillow unavailable
DEBUG:BlenderGIS-master.core.checkdeps:ImageIO Freeimage plugin available
DEBUG:BlenderGIS-master.operators.view3d_mapviewer:Bounding box request : (-223386909.41531447, -145037120.93432996, 223386909.41531447, 145037120.93432996)
DEBUG:BlenderGIS-master.core.basemaps.mapservice:117 tiles requested, 0 already in cache, 117 remains to download
DEBUG:BlenderGIS-master.core.basemaps.mapservice:http://mt0.google.com/vt/lyrs=s&x=0&y=0&z=0
ERROR:BlenderGIS-master.core.basemaps.mapservice:Corrupted tile on cache
Traceback (most recent call last):
  File "/Users/laurentb/Library/Application Support/Blender/3.5/scripts/addons/BlenderGIS-master/core/basemaps/mapservice.py", line 883, in getImage
    img = NpImage(data)
  File "/Users/laurentb/Library/Application Support/Blender/3.5/scripts/addons/BlenderGIS-master/core/georaster/npimg.py", line 122, in __init__
    self.data = self._npFromBLOB(data)
  File "/Users/laurentb/Library/Application Support/Blender/3.5/scripts/addons/BlenderGIS-master/core/georaster/npimg.py", line 242, in _npFromBLOB
    data = self._npFromImgIO(img)
  File "/Users/laurentb/Library/Application Support/Blender/3.5/scripts/addons/BlenderGIS-master/core/georaster/npimg.py", line 258, in _npFromImgIO
    data = imageio.imread(img)
  File "/Users/laurentb/Library/Application Support/Blender/3.5/scripts/addons/BlenderGIS-master/core/lib/imageio/core/functions.py", line 181, in imread
    reader = read(uri, format, 'i', **kwargs)
  File "/Users/laurentb/Library/Application Support/Blender/3.5/scripts/addons/BlenderGIS-master/core/lib/imageio/core/functions.py", line 107, in get_reader
    raise ValueError('Could not find a format to read the specified file '
ValueError: Could not find a format to read the specified file in mode 'i'

**Comment by stabelchod** [comment]
Working also on Blender 3.6 for Apple Silicon. Thanks!

> brew install freeimage
> launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib

**Comment by the-mitacek** [comment]
Hi, can anyone help, how to use this hack? GIS not working on my M1 MacOS 13.4, Blender 3.6.0

Thanks!

> brew install freeimage
> launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib

**Comment by vavdav** [comment]
> Hi, can anyone help, how to use this hack? GIS not working on my M1 MacOS 13.4, Blender 3.6.0
> 
> Thanks!
> 
> > brew install freeimage
> > launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib

you just need terminal and homebrew (package installer) 
https://brew.sh/

Then just run those commands.  You have to check that `/opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib` path exists. It could be different version.

**Comment by multitudes** [comment]
AlsoI I had to check my environment to make it work..and sometimes launching blender from the terminal produced different results!

**Comment by nirruk** [comment]
> > Hi, can anyone help, how to use this hack? GIS not working on my M1 MacOS 13.4, Blender 3.6.0
> > Thanks!
> > > brew install freeimage
> > > launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib
> 
> you just need terminal and homebrew (package installer) https://brew.sh/
> 
> Then just run those commands. You have to check that `/opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib` path exists. It could be different version.

umm.. here's another MacBook pro M1 (ver. 12.6.7), and I'm literally a newbie of these things..(still struggling for hours ;-;)
Can you show me the process by something like a photo or tutorial video?

**Comment by vvrakking** [comment]
> ### TL;DR
> 
> imageio lib is outdated, temp. workaround:
> 
> ```
> brew install freeimage
> launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib
> ```
> 
> _running on macOS 12.3, blender 3.2.1, blendergis 2.2.8_
> 
> cheers.
> ### Long story
> 
> So, I dug in a bit and found, that the error message was a symptom, but not the root cause. The original error was shadowed, and is thrown in thrown [here](https://github.com/domlysz/BlenderGIS/blob/7acdaae88ddcb43345daca9e4b310fd8bc626ce4/core/lib/imageio/plugins/_freeimage.py#L408). This error simply means: “The freeimage library could not be loaded.” So, the imageio library can't find the freeimage library it depends upon. Also, it can't download the library, as the version of freeimage is outdated and does not know about the existence of an arm macos. So even if you have installed freeimage on your system, imageio won’t find it.
> 
> So, what can we do? Luckily, I found [this block of code](https://github.com/domlysz/BlenderGIS/blob/7acdaae88ddcb43345daca9e4b310fd8bc626ce4/core/lib/imageio/plugins/_freeimage.py#L465), which gave hope.
> 
> ```
> # Add the path specified with IMAGEIO_FREEIMAGE_LIB:
> lib = os.getenv('IMAGEIO_FREEIMAGE_LIB', None)
> if lib is not None:
>     exact_lib_names.insert(0, lib)
> ```
> 
> I installed the freeimage library using `brew install freeimage`, there’s an arm build also.
> 
> Then, setting the env var in the terminal and starting blender did the job (yay 🥳). It’s not the most convenient solution, as you always have to start blender from the terminal or some custom script, even if you set the env var in `~/.zshrc` or something.
> 
> Fortunately, you can set env vars for gui applications with `launchctl`, so it’s as easy as running the following:
> 
> ```
> launchctl setenv IMAGEIO_FREEIMAGE_LIB /opt/homebrew/Cellar/freeimage/3.18.0/lib/libfreeimage.dylib
> ```
> 
> Just make sure you’re entering the correct version, it was `3.18.0` at the time of writing.
> 
> cheers again.

Worked for me on Mac Studio - MacOS 14.2.1, Blender 4.0.2. with BlenderGIS 2.2.8. Thank you very much!

**Comment by airbag00** [comment]
This worked for me!  blender 4.1 on a M1 mac mini running macos Sonoma 14.0. 
Thank you very much!!

**Comment by nurlopez** [comment]
@aplr amazing! 🙌

This workaround worked smoothly for me: M1 MacOS 14.1.1 and Blender 4.1.1 🎉

**Comment by k05yra** [comment]
"I've installed both Homebrew and FreeImage on my macOS 14.2.1, ensuring that the paths are correct. I'm running Blender 4.2.1 with a topography API key. However, whenever I load a basemap, all I see is [describe what you see on the screen, such as a blank map, distorted image, etc.]. Sometimes, the entire application freezes and crashes. Any advice on how to fix this?" 
<img width="1440" alt="Screenshot 2024-08-21 at 4 13 05 AM" src="https://github.com/user-attachments/assets/c598d476-f9d2-4cfa-a511-6cf5e4c033bd">

**Comment by k05yra** [comment]
I've installed both homebrew and freeimage on my macOS 14.2.1, ensuring that the paths are correct. I'm running Blender 4.2.1 with a topography API key. However, whenever I load a basemap, all I see is what's pictured below. Sometimes, the entire application freezes and crashes. Any advice on how to fix this?

<img width="1440" alt="Screenshot 2024-08-21 at 4 13 05 AM" src="https://github.com/user-attachments/assets/886c970b-d852-42df-94c0-370eff9bab64">
